Docs for Immersive Intelligence mod (1.12.2) (#503)
CraftTweaker/CraftTweaker-Documentation: docs/1.12/content/Mods/Immersive_Intelligence/CraftTweaker_Support/Chemical_Painter.md

::requiredMod[Immersive Intelligence]{builtIn=true
modLink=https://www.curseforge.com/minecraft/mc-mods/immersive-intelligence}

# Chemical Painter

> The `ChemicalPainter` package can be used to add/remove recipes to/from the II Chemical Painter.

Reviewer: ```suggestion
The `ChemicalPainter` package can be used to modify the Immersive Intelligence Chemical Painter recipes.
```

## Importing the Package

```zenscript
import mods.immersiveintelligence.ChemicalPainter;
```

## Adding Recipes

### Parameters

| Type                                                | Name         | Required |
|-----------------------------------------------------|--------------|----------|
| [IIngredient](/Vanilla/Variable_Types/IIngredient/) | Item Input   | Yes      |
| [IItemStack](/Vanilla/Items/IItemStack/)            | Item Output  | Yes      |
| int                                                 | Paint Amount | Yes      |
| int                                                 | Energy       | Yes      |
| int                                                 | Time         | Yes      |

> Paint amount is the total amount of ink in mB used for a single process.  
> Amounts of specific ink are based on their percentage in the [CMYK](https://en.wikipedia.org/wiki/CMYK_color_model)
> color.

Reviewer: ```suggestion
Paint amount is the total amount of ink in mB used for a single process.  
Amounts of specific ink are based on their percentage in the [CMYK](https://en.wikipedia.org/wiki/CMYK_color_model)
color model.
```

### Syntax

```zenscript
mods.immersiveintelligence.ChemicalPainter.addRecipe(IIngredient itemInput, IChemicalPainterFunction function, int paintAmount, int energy, int time);
```

### IChemicalPainterFunction

> IChemicalPainterFunction is a function used to determine the item outputted by a recipe based on the stack and selected

Reviewer: ```suggestion
IChemicalPainterFunction is a function used to determine the output item of a recipe based on the stack and selected
```

paint color.    

> It can be used i.e. to get core NBT tag or merge core and casing's NBT tags.

Reviewer: ```suggestion
For example, it can be used to get the core NBT tag or merge the core and casing's NBT tags.
```
